packages/compiler-vapor/src/generators/for.ts

```diff
@@ -27,6 +27,8 @@ export function genFor(
 
   const sourceExpr = ['() => (', ...genExpression(source, context), ')']
   let updateFn = '_updateEffect'
+
+  vForGenEffects.push(genEffectInFor)
   context.genEffect = genEffectInFor
```

Comment:
Let's remove `genEffect` and use `vForGenEffects` only (call the last one). Also, rename it to `genEffects`